### PR TITLE
feat(test): run tests and update build result

### PR DESCRIPTION
**Pull request by JohelEGP**
- [x] More field testing.
- [x] Copy to the repository results that differ from the previous compiler version only.

### CODE REVIEW

File: regression-tests/CMakeLists.txt [hunk 1]
```diff
@@ -119,8 +146,48 @@ function(cppfront_build_tests)
         "-Dcppfront_DIR=${cppfront_DIR}"
         "-Dcppfront-exe_DIR=${cppfront-exe_DIR}"
         "-DCPPFRONT_FLAGS=${ARG_EXTRA_FLAGS}"
+        # There's `CMAKE_CXX_LINKER_LAUNCHER`, too. So far, it's not needed.
+        "-DCMAKE_CXX_COMPILER_LAUNCHER=${CMAKE_COMMAND};-D;OUTPUT_FILE=${gen_cpp_src}.output;-P;../../ExecuteWithRedirection.cmake;--"
```
Comment (JohelEGP) [inline regression-tests/CMakeLists.txt:148]: These only work with "The Makefile Generators and the Ninja generator".
The alternative is [`CMAKE_EXPORT_COMPILE_COMMANDS`][].
I already handle that at <https://github.com/JohelEGP/jegp.cmake_modules/blob/master/modules/.detail/JEGPReadExportedCompileCommand.cmake>.

[`CMAKE_EXPORT_COMPILE_COMMANDS`]: https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html

Comment (JohelEGP) [reply regression-tests/CMakeLists.txt:158]: I would still need to build and install.
So I'd compile twice to get the output.
Maybe I can hijack `CMAKE_CXX_COMPILER`
and use it like `CMAKE_CXX_COMPILER_LAUNCHER`.

**Comment by alexreinking** [comment]
> All this would be much simpler if we could just work in the Git repository itself rather than on copies of its files.

I agree, but @hsutter didn't want to include a CMake build in the upstream cppfront, at least not yet. See the discussion here: https://github.com/hsutter/cppfront/pull/15#issuecomment-1264487351

It has been some time now, though -- perhaps Herb would be willing to reconsider?

**Comment by alexreinking** [comment]
Also, is this ready to review?

**Comment by JohelEGP** [comment]
> Also, is this ready to review?

Yes, since <https://github.com/modern-cmake/cppfront/pull/88#event-9489951378>.
I still continue to find rough edges when developing.

> > All this would be much simpler if we could just work in the Git repository itself rather than on copies of its files.
> 
> I agree, but @hsutter didn't want to include a CMake build in the upstream cppfront, at least not yet. See the discussion here: [hsutter/cppfront#15 (comment)](https://github.com/hsutter/cppfront/pull/15#issuecomment-1264487351)
> 
> It has been some time now, though -- perhaps Herb would be willing to reconsider?

I was hoping we wouldn't be blocked on that.

I've also found the need to manually specify the tests' build type as `Debug` because it's hard-coded.
So it would be much better if there was a single CMake project,
even if we can't have subdirectories within the cppfront git submodule.

**Comment by alexreinking** [comment]
Hey, sorry I took so long to get back to this -- merging now...

**Pull request merged**